### ISSUE TITLE
Error when a method is passed as an endpoint 

### ISSUE THREAD

**Issue by taoufik07** [kennethreitz/responder#126]
When a method is passed as an endpoint and we try to add `is_routed` to it, an attribute error is raised and it's passed it silently, So the `route.is_function` fails and that's what caused schemas to fail since the endpoint is a mehtod `API.schema_response`.

**UPDATE**

One of the solutions : 

```
@property
def is_function(self):
    # We can also remove is_routed ?
    routed = hasattr(self.endpoint, "is_routed") or callable(self.endpoint)
    code = hasattr(self.endpoint, "__code__")
    kwdefaults = hasattr(self.endpoint, "__kwdefaults__")
    return all((routed, code, kwdefaults))
```
or : 
```
try:
    if callable(endpoint):
        endpoint.__dict__["is_routed"] = True
except TypeError:
    pass
```

@kennethreitz  What solution you prefer ?

**Comment by taoufik07** [comment]
You can now close this issue since you've merged #131

**Comment by taoufik07** [comment]
Need to be reopened since you reverted :3
